### PR TITLE
add some inline comments

### PR DESCRIPTION
**Pull request by DJViau**
## Motivation

Adds inline comments, mostly to the reference implementations, for the sake of clarity.

## Solution

I may have relied on GPT3 and GitHub Copilot a bit too heavily in my effort to move quickly and make up for my lack of understnading, please make sure that these make sense.

### CODE REVIEW

File: contracts/lib/OrderValidator.sol [hunk 1]
```diff
@@ -369,7 +374,8 @@ contract OrderValidator is Executor, ZoneInteraction {
                 orderParameters.offer = extendedOffer;
             }
 
-            // Loop through offer and ensure at least as much on returned offer
+            // Loop through each offer and ensure at least as much on returned
```
Comment (BCLeFevre) [inline contracts/lib/OrderValidator.sol:377]: 🤔 
This sounds weird.

Comment (DJViau) [reply contracts/lib/OrderValidator.sol:377]: It was written by 0 and I was scared to materially change it, so I just tried to make it a wee bit clearer 😅

Comment (jameswenzel) [reply contracts/lib/OrderValidator.sol:377]: "Loop through each new offer and ensure the new amounts are at least as much as the respective original amounts"?

Comment (DJViau) [reply contracts/lib/OrderValidator.sol]: Sounds good.  Updated

**Pull request merged**